### DISCUSSION TITLE
How to save local changes? (Opposite of apply)

### DISCUSSION THREAD

**Discussion by halloleo** [twpayne/chezmoi#3799] [Q&A] [answered]
I think I miss something basic here: 

When I have changed my local file `~/.bashrc` (I think you call it the "target" file?) and then want to save the changed file to chezmoi's store (so that on another machine I can use it), how do I do this? 

`apply` seem to only be able to restore the file from chezmoi's store.

**Comment by twpayne** [comment, marked as answer]
If you've made changes to the target file you can run [`chezmoi add`](https://www.chezmoi.io/reference/commands/add/) to add the file again, or [`chezmoi re-add`](https://www.chezmoi.io/reference/commands/re-add/) to add all managed files again. However, you should probably re-familiarize yourself with the [quick-start guide](https://www.chezmoi.io/quick-start/) and [this FAQ entry](https://www.chezmoi.io/user-guide/frequently-asked-questions/usage/#how-do-i-edit-my-dotfiles-with-chezmoi).

**Reply by halostatue** [reply]
I also strongly recommend learning about [chezmoi merge](https://www.chezmoi.io/reference/commands/merge/) and [chezmoi merge-all](https://www.chezmoi.io/reference/commands/merge-all/) if (like me) you’re going to regularly edit the files external to chezmoi.

**Reply by halloleo** [reply]
Thanks for the details, @twpayne! I certainly looked at the quick start guide, but because I don't use the git part I couldn't see how to save back the changed target file. The FAQ entry makes it clear.

@halostatue Yes, `merge` looks very helpful - just have to set it up so that the merge gets done in Emacs.
